modular_skyrat/modules/primitive_fun/code/glassblowing.dm

```diff
@@ -301,42 +300,49 @@
 						to_chat(user, span_warning("You interrupt an action!"))
 						in_use = FALSE
 						return
-					if(!check_valid_table())
+					if(!check_valid_table(user))
 						to_chat(user, span_warning("You must be near a non-flammable table!"))
 						in_use = FALSE
 						return
 					find_glass.current_actions[2]++
 					to_chat(user, span_notice("You finish spinning [src]."))
+					user.mind.adjust_experience(/datum/skill/production, 10)
 				if("Remove")
 					if(find_glass.current_actions[1] < find_glass.required_actions[1])
 						in_use = FALSE
 						find_glass.forceMove(get_turf(src))
+						icon_state = "blow_pipe_empty"
 						return
 					if(find_glass.current_actions[2] < find_glass.required_actions[2])
 						in_use = FALSE
 						find_glass.forceMove(get_turf(src))
+						icon_state = "blow_pipe_empty"
```

Comment:
could these three lines (
```
						in_use = FALSE
						find_glass.forceMove(get_turf(src))
						icon_state = "blow_pipe_empty"
```
) be converted into a proc for less code duplication?